File: Clean-Architecture/Feature/dart/example/lib/features/auth/presentation/controllers/auth_controller.dart
import 'package:flutter/foundation.dart';

import '../../domain/entities/auth_session.dart';
import '../../domain/usecases/login_user.dart';

class AuthController extends ChangeNotifier {
  AuthController(this._loginUser);
  final LoginUser _loginUser;

  bool isLoading = false;
  String? errorMessage;
  AuthSession? session;

  Future<bool> login({required String email, required String password}) async {
    isLoading = true;
    errorMessage = null;
    notifyListeners();

    final result = await _loginUser(email: email, password: password);
    final didSucceed = result.fold(
      onFailure: (failure) {
        errorMessage = failure.message;
        return false;
      },
      onSuccess: (value) {
        session = value;
        return true;
      },
    );

    isLoading = false;
    notifyListeners();
    return didSucceed;
  }

  void logout() {
    session = null;
    errorMessage = null;
    notifyListeners();
  }
}

File: Clean-Architecture/Feature/Javascript/example/src/features/auth/presentation/controllers/AuthController.js
import { validateRequired } from '../../../../core/validation/validateRequired.js';

export class AuthController {
  constructor({ loginUser }) {
    this.loginUser = loginUser;
  }

  login = async (req, res) => {
    validateRequired(req.body, ['email', 'password']);
    const result = await this.loginUser.execute(req.body);
    res.status(200).json({ data: result });
  };
}
